server/tests/test_input_rail_adversarial.py

```
@@ -0,0 +1,228 @@
+
+class _StubRails:
+    """Mimics enough of nemoguardrails.LLMRails for input_rail.check_input.
+
+    Treats messages mentioning the literal token ``__BLOCK__`` as a triggered
+    rail; everything else passes.
+    """
+
+    def __init__(self, raise_for: str | None = None) -> None:
+        self._raise_for = raise_for
+
+    async def generate_async(self, *, messages, options=None):  # noqa: D401
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `generate_async`

(ANN202)